pkg/graph/resolvers/plan_collaborator.go

```diff
@@ -4,23 +4,83 @@ import (
 	"github.com/google/uuid"
 	"go.uber.org/zap"
 
+	"github.com/cmsgov/mint-app/pkg/email"
+	"github.com/cmsgov/mint-app/pkg/shared/oddmail"
+
 	"github.com/cmsgov/mint-app/pkg/authentication"
 	"github.com/cmsgov/mint-app/pkg/graph/model"
 	"github.com/cmsgov/mint-app/pkg/models"
 	"github.com/cmsgov/mint-app/pkg/storage"
 )
 
 // CreatePlanCollaborator implements resolver logic to create a plan collaborator
-func CreatePlanCollaborator(logger *zap.Logger, input *model.PlanCollaboratorCreateInput, principal authentication.Principal, store *storage.Store) (*models.PlanCollaborator, error) {
+func CreatePlanCollaborator(
+	logger *zap.Logger,
+	emailService oddmail.EmailService,
+	emailTemplateService email.TemplateService,
+	input *model.PlanCollaboratorCreateInput,
+	principal authentication.Principal,
+	store *storage.Store) (*models.PlanCollaborator, error) {
 	collaborator := models.NewPlanCollaborator(principal.ID(), input.ModelPlanID, input.EuaUserID, input.FullName, input.TeamRole, input.Email)
-
 	err := BaseStructPreCreate(logger, collaborator, principal, store, true)
 	if err != nil {
 		return nil, err
 	}
 
 	retCollaborator, err := store.PlanCollaboratorCreate(logger, collaborator)
-	return retCollaborator, err
+	if err != nil {
+		return nil, err
+	}
+
+	modelPlan, err := store.ModelPlanGetByID(logger, input.ModelPlanID)
+	if err != nil {
+		return nil, err
+	}
+
+	err = sendCollaboratorAddedEmail(emailService, emailTemplateService, input.Email, modelPlan)
+	if err != nil {
+		return nil, err
+	}
+
+	return retCollaborator, nil
+}
+
+func sendCollaboratorAddedEmail(
+	emailService oddmail.EmailService,
+	emailTemplateService email.TemplateService,
+	receiverEmail string,
+	modelPlan *models.ModelPlan,
+) error {
+	if emailService == nil || emailTemplateService == nil {
+		return nil
+	}
+
+	emailTemplate, err := emailTemplateService.GetEmailTemplate(email.AddedAsCollaboratorTemplateName)
+	if err != nil {
+		return err
+	}
+
+	emailSubject, err := emailTemplate.GetExecutedSubject(email.AddedAsCollaboratorSubjectContent{
+		ModelName: modelPlan.ModelName,
+	})
+	if err != nil {
+		return err
+	}
+
+	emailBody, err := emailTemplate.GetExecutedBody(email.AddedAsCollaboratorBodyContent{
```

Comment:
I think we might need to add `CLIENT_PROTOCOL` to the link because without it the link from the email is relative, so it'll link to localhost:1025/model-plan/etc...

Reply:
This is a great point. Rather than formatting our client protocol along with hostname I have swapped to using the CLIENT_ADDRESS env var to address both issues.